internal/transform/transform.go

```diff
@@ -118,6 +176,9 @@ func ExtractStyles(doc *astro.Node) {
 
 func ExtractScript(doc *astro.Node, n *astro.Node) {
 	if n.Type == astro.ElementNode && n.DataAtom == a.Script {
+		if HasSetDirective(n) {
+			return
+		}
```

Comment:
Per the RFC, the existence of `set:html` or `set:text` means the `<script>` will be printed literally